projects/drivers/xilinx_platform/platform_drivers.c

```diff
@@ -97,21 +132,18 @@ int32_t i2c_write(struct i2c_desc *desc,
 		  uint8_t bytes_number,
 		  uint8_t stop_bit)
 {
-	if (desc) {
-		// Unused variable - fix compiler warning
-	}
-
-	if (data) {
-		// Unused variable - fix compiler warning
-	}
+	u8 teminate_option;
+	int32_t ret;
 
-	if (bytes_number) {
-		// Unused variable - fix compiler warning
-	}
+	if(stop_bit == 1)
+		teminate_option = XIIC_STOP;
+	else
+		teminate_option = XIIC_REPEATED_START;
 
-	if (stop_bit) {
-		// Unused variable - fix compiler warning
-	}
+	ret = XIic_Send(desc->instance.BaseAddress, desc->slave_address, data,
+			bytes_number, teminate_option);
```

Comment:
Since teminate_option is used only once, I would remove it and use something like:
stop_bit ? XIIC_STOP : XIIC_REPEATED_START
Similar for i2c_read().